### PR TITLE
chore: bump golang-jwt to v5

### PR DESCRIPTION
**Pull request by David-Wobrock**
The rationale behind this update is that we see an ever increasing memory usage on our Oathkeeper instances, until they OOM and we start a new one.

From some pprof analysis in our prod environment, seems to indicate that it's in `MutatorIDToken.Mutate` that the issue lies.
More specifically when signing the token string.
- in `MutatorIDToken.Mutate` => `signed, err := a.r.CredentialsSigner().Sign(r.Context(), jwks, claims)`
- in `Sign` => `signed, err := token.SignedString(key.Key)`
- in `SignedString` => `return strings.Join([]string{sstr, sig}, "."), nil`

Screenshot 2024-06-25 at 17 26 37

In golang-jwt v5, this **method does not use `strings.Join` anymore**.
See https://github.com/golang-jwt/jwt/pull/115

I'm entirely sure to understand why string joining could cause this issue, perhaps some reference that Oathkeeper is keeping and therefore leads to this memory leak 🤷 
But in any case, we believe it's a good thing to keep dependencies up-to-date and use the latest version.

Let me know what you think :) 

## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ ] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got the approval (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](https://github.com/ory/docs).

### CODE REVIEW

**Review by aeneasr — COMMENTED** [review]
Very nice! I don't think it will solve the mem leak issue but good to see.

The mem leak might come from an incorrect cache configuration for the ID token mutator.

File: credentials/verifier_default.go [hunk 1]
```diff
@@ -82,15 +82,8 @@ func (v *VerifierDefault) Verify(
 		}
 
 		return nil, errors.WithStack(herodot.ErrBadRequest.WithReasonf(`The signing key algorithm does not match the algorithm from the token header.`))
-	})
+	}, jwt.WithIssuedAt())
 	if err != nil {
-		if e, ok := errors.Cause(err).(*jwt.ValidationError); ok {
```
[the remark this answers is not in the document]
Comment (David-Wobrock) [reply credentials/verifier_default.go:87]: It has been deleted in v5.
See https://github.com/golang-jwt/jwt/commit/148d71010923ca691c950db9846191800f498f8d#diff-76a7e0e299c7417bae32d3098e71370980157fe29e68a366671491d147d40572L51-L55

The original PR: https://github.com/golang-jwt/jwt/pull/274
And instead, the project is using plain `error` types from my understanding.

[the remark this answers is not in the document]
Comment (aeneasr) [reply credentials/verifier_default.go:87]: Ok, but then this needs to be replaced with `errors.Is(err, jwt.WhatEverErrorTypeIsTheCorrectOne)`

[the remark this answers is not in the document]
Comment (David-Wobrock) [reply credentials/verifier_default.go:87]: I've pushed an edit adding a condition with all errors that the `ValidationError` was potentially carrying with it.

**Comment by David-Wobrock** [comment]
> Very nice! I don't think it will solve the mem leak issue but good to see the update.
> 
> The mem leak might come from an incorrect cache configuration for the ID token mutator.

You're right, profiling locally I have the same memory usage on `master` and this branch.

What obviously helps, is to disable the cache 🙄 
But generally, I think that making the `id_token` mutator cache configurable is probably a good move.
Both enable/disable, but also configure the max cost/tokens - like the authn OAuth2 ClientCredentials and introspection handlers.

I'll try to suggest a PR that adds these options.

**Review by aeneasr — APPROVED** [review]

**Pull request merged**